sdk/tests/conformance/extensions/webgl-webcodecs-video-frame.html

```diff
@@ -0,0 +1,199 @@
+      ctx.fillStyle = src_color;
+      let drawOneFrame = function (time) {
+        ctx.fillStyle = src_color;
+        ctx.fillRect(0, 0, src_width, src_height);
+        window.requestAnimationFrame(drawOneFrame);
+      }
+      window.requestAnimationFrame(drawOneFrame);
+    }
+
+    function captureAndEncode(processChunk) {
+      let cnv = document.getElementById("src");
+      let fps = 60;
+      let pending_outputs = 0;
+      let stream = cnv.captureStream(fps);
+      let vtr = new VideoTrackReader(stream.getVideoTracks()[0]);
+
+      const init = {
+        output: (chunk) => {
+          testPassed("Encode frame successfully.");
+          pending_outputs--;
+          processChunk(chunk);
+        },
+        error: (e) => {
+          testFailed("Failed to encode frame.");
+          finishTest();
+          vtr.stop();
+        }
+      };
+
+      const config = {
+        codec: codec_string,
+        width: cnv.width,
+        height: cnv.height,
+        bitrate: 10e6,
+        framerate: fps,
+      };
+
+      let encoder = new VideoEncoder(init);
+      encoder.configure(config);
+
+      vtr.start((frame) => {
+        if (pending_outputs > 30) {
+          console.log("drop this frame");
+          // Too many frames in flight, encoder is overwhelmed
+          // let's drop this frame.
+          return;
+        }
+        if (frame_counter == kMaxFrame) {
+          return;
+        }
+
+        frame_counter++;
+        pending_outputs++;
+        const insert_keyframe = (frame_counter % 150) == 0;
+        encoder.encode(frame, { keyFrame: insert_keyframe });
+      });
+    }
+
+    function startDecodingAndRendering(cnv, handleFrame) {
+      const init = {
+        output: handleFrame,
+        error: (e) => {
+          testFailed("Failed to decode frame.");
+          finishTest();
+        }
+      };
+
+      const config = {
+        codec: codec_string,
+        codedWidth: cnv.width,
+        codedHeight: cnv.height,
+        acceleration: "deny",
+
+      };
+
+      let decoder = new VideoDecoder(init);
+      decoder.configure(config);
+      return decoder;
+    }
+
+    function isFramePixelMatched(gl, th_per_pixel = pixelCompareTolerance) {
+      WebGLTestUtils.checkCanvasRect(gl, 0, 0, src_width, src_width, kTestPixel, "should be orange", pixelCompareTolerance)
+    }
+
+    function main() {
+      if (!("VideoEncoder" in window)) {
+        testPassed("WebCodecs API is not supported.");
+        finishTest();
+        return;
+      }
+      let cnv = document.getElementById("dst");
+
+      let webgl_webcodecs_test_context = {
+        maxFrameTested: kMaxFrame,
+        displayed_frame: 0,
+        isFramePixelMatched: isFramePixelMatched,
+        testFailed: testFailed,
+        testPassed: testPassed,
+        finishTest: finishTest
+      };
+      setTestMode(webgl_webcodecs_test_context);
+      let handleFrame = requestWebGLVideoFrameHandler(cnv);
+      if (handleFrame === null) {

```

Comment:
Would be good if this used the testPassed primitive to log the output. console.log isn't reflected easily in all of the automated test harnesses that use the WebGL conformance tests.